### ISSUE TITLE
[RF] RooFit should give an error message when using different variables with same name

### ISSUE THREAD

**Issue by lmoneta** [root-project/root#7417]
If one builds a model where two variables have the same name, everything go un-noticed and result can be not correct. 
An example is in the test program attached to ROOT-3555, where a simultaneous pdf is built with two Gaussian, defined with different mean and sigma parameters, having the same name.

**Comment by Harshalzzzzzzz** [comment]
@lmoneta Dear Mentor,
I'm interested in contributing to the RooFit and I would like to work on this issue. Can you please elaborate more on what needs to be done and I can submit a Pull Request for the same?

Thanks!

**Comment by lmoneta** [comment]
Hello @Harshalzzzzzzz !

In this case we need to inspect the model content and when we find two different nodes with the same name we emit a warning or an error message. 
This could be done when calling `pdf.fitTo`, i.e. before creating the RooNLLVar object or during its creation. 
See https://github.com/root-project/root/blob/master/roofit/roofitcore/src/RooAbsPdf.cxx#L1292

For inspecting the computational tree graph, you can look at functions like `RooAbsArg::servers()` or `RooAbsArg::getComponets`

Thank you for your offer to contribute to RooFit

Lorenzo

If you want to contribute, we have maybe tis old issue, which is easier to fix and it is has some relation to the work you have done on the Pythonization. See https://sft.its.cern.ch/jira/browse/ROOT-2784

**Comment by Harshalzzzzzzz** [comment]
@lmoneta Do I need to check cmdList parameter for same name and then emit a warning/error about the same?

**Comment by lmoneta** [comment]
Yes, this is correct, in case we want to fix https://sft.its.cern.ch/jira/browse/ROOT-2784
For this issue, #7417, it is different

**Comment by ynikitenko** [comment]
"when we find two different nodes with the same name we emit a warning or an error message" - I'd like to add that when I make a fit with many pdfs (with many warnings and messages), such a warning can be easily unnoticed, while the results will be wrong (see the previous link).

I vote for a C++ exception in this case and a loud failure as early as possible (but it's up to you, because I can't imagine all cases for RooFit).
